packages/fxa-admin-server/src/lib/version.ts

```diff
@@ -0,0 +1,73 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/*
+ * Return version info based on package.json, the git sha, and source repo
+ *
+ * Try to statically determine commitHash and sourceRepo at startup.
+ *
+ * If commitHash cannot be found from `version.json` (i.e., this is not
+ * production or stage), then an attempt will be made to determine commitHash
+ * and sourceRepo dynamically from `git`. If it cannot be found with `git`,
+ * just show 'unknown' for commitHash and sourceRepo.
+ *
+ * This module may be called as ./dist/lib/version.js, or when testing,
+ * ./lib/version.ts so we need to look for `package.json` and `version.json`
+ * in two possible locations.
+ */
+
+import cp from 'child_process';
+import path from 'path';
+
+function readJson(filepath: string) {
+  try {
+    return require(filepath);
+  } catch (e) {
+    /* ignore */
+  }
+
+  return;
+}
+
+function getValue(name: string, command: string): string {
+  const value =
+    readJson(path.resolve(__dirname, '..', 'version.json')) ||
+    readJson(path.resolve(__dirname, '..', '..', 'version.json'));
+
+  if (value?.version?.[name]) {
+    return value.version[name];
+  }
+
+  let stdout = 'unknown';
+  try {
+    stdout = cp.execSync(command, { cwd: __dirname }).toString('utf8');
+  } catch (e) {
+    /* ignore */
+  }
+
+  return stdout?.toString().trim();
+}
+
+interface Version {
+  commit: string;
+  source: string;
+  version: string;
+}
+
+function getVersionInfo(): Version {
+  const commit = getValue('hash', 'git rev-parse HEAD');
+  const source = getValue('source', 'git config --get remote.origin.url');
+
+  const packageInfo =
+    readJson(path.resolve(__dirname, '..', '..', 'package.json')) ||
+    readJson(path.resolve(__dirname, '..', '..', '..', 'package.json'));
+
+  return {
+    commit,
+    source,
+    version: packageInfo?.version
+  };
+}
+
+export const version = getVersionInfo();
```

Comment:
Is this file from somewhere?  I don't see any tests for, but if it's working code from somewhere 🤷‍♂

Reply:
It is, its from event-broker which is from auth-server. Ideally it should be in fxa-shared.